src/blocking/body.rs

```diff
@@ -248,7 +248,7 @@ async fn send_future(sender: Sender) -> Result<(), crate::Error> {
                 buf.reserve(8192);
             }
 
-            match body.read(unsafe { buf.bytes_mut() }) {
+            match body.read(&mut buf) {
```

Comment:
This won't do the write thing: `&mut BytesMut` will just access the previously written-to bytes, and we need the tail end, thus `bytes_mut()`.